### PR TITLE
(#160) Improve messages in Assertion and Mismatches

### PR DESCRIPTION
**Pull request by victornoel**
This is for #160. I tried to cleanup a bit how Assertion and Mismatches behave and so that it's consistent with all the matchers.

### CODE REVIEW

**Comment by victornoel** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/llorllale/cactoos-matchers/issues/231#issuecomment-779362975))

@victornoel This is what I know about this job in C63314D6Z, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#231](https://github.com/llorllale/cactoos-matchers/issues/231) is [in scope](http://www.zerocracy.com/policy.html#14) for 22hr
  * The role is `REV`
  * The job is assigned to @baudoliver7[/z](https://www.0crat.com/u/baudoliver7) for [21hr](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @victornoel[/z](https://www.0crat.com/u/victornoel): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Allorllale%2Fcactoos-matchers%23231%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Review by baudoliver7 — CHANGES_REQUESTED** [review]
@victornoel Please, see comments below.

File: src/test/java/org/llorllale/cactoos/matchers/MismatchesTest.java [hunk 1]
```diff
@@ -54,80 +50,76 @@ void example() {
...
         ).affirm();
     }
 
     @Test
-    void matches() {
+    void mismatchesWithWrongMessage() {
         new Assertion<>(
-            "Must fail to mismatch",
-            new Mismatches<>(new TextOf("a"), new TextOf("expected")),
+            "Must mismatches",
```
[the remark this answers is not in the document]
Comment (victornoel) [reply src/test/java/org/llorllale/cactoos/matchers/MismatchesTest.java]: @baudoliver7 arg, again :) thx!

File: src/main/java/org/llorllale/cactoos/matchers/HasValues.java [hunk 2]
```diff
@@ -65,7 +65,9 @@ public HasValues(final Iterable<X> expected) {
                 actual -> new ListOf<>(actual).containsAll(new ListOf<>(expected)),
                 desc -> desc.appendText("contains ")
                     .appendValue(new TextOf(expected)),
```
[the remark this answers is not in the document]
Comment (victornoel) [reply src/main/java/org/llorllale/cactoos/matchers/HasValues.java]: @baudoliver7 good find, and I looked around and didn't find any similar thing in the codebase too :)

**Comment by victornoel** [comment]
@baudoliver7 it should be ok now

**Comment by baudoliver7** [comment]
@victornoel Thanks, it's ok for me :)

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @baudoliver7[/z](https://www.0crat.com/u/baudoliver7), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**